sas-viya/azuredeploy.parameters.json

```diff
@@ -0,0 +1,24 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "DeploymentDataLocation" : {
+      "value": "https://azureviya.blob.core.windows.net/mercury-deployment-data/TravisCI/TravisLicense34.zip?sp=r&st=2019-02-14T21:16:19Z&se=2019-02-20T05:16:19Z&spr=https&sv=2018-03-28&sig=zM2HpnhM46yz9lvX4x0h0f4n5NY8BdfR1kHxMxseYHA%3D&sr=b"
```

Comment:
I am afraid we cannot make this change. This is a reference to a temporary copy of a valid SAS license file for the purposes of running through TavisCI. Customers are expected to bring their own valid license from SAS in order to run this and we pull down the TravisCI license as soon as the build passes TravisCI.

Reply:
Assume there's no trial license then?  Users will use this value and fail (happens all the time in this repo)...

Reply:
At this time, sas does not offer trial licenses. If there is a way to both pass TravisCI and clearly indicate that this is a dummy value they need to replace, I would be happy to implement that, but otherwise the notes in the readme.md on the subject are the best we can do.

Reply:
ok, let's add a metadata tag to the parameter that mentions this - e.g. this is a test value, you must supply a valid license file for deployment to succeed...

Reply:
When you say add a metadata tag to the parameter, do you mean add it to the existing help description in azuredeploy.json or to the parameter in azuredeploy.parameters.json? If the latter, would you happen to have a reference for how to do that? Trying to just add a similar metadata/description block to the parameter in the azuredeploy.parameters.json file results in an error:

Azure Error: InvalidRequestContent
Message: The request content was invalid and could not be deserialized: 'Could not find member 'metadata' on object of type 'DeploymentParameterDefinition'. Path 'properties.parameters.DeploymentDataLocation.metadata', line 1, position 48319.'.

Reply:
in the parameters file, eg

{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "DeploymentDataLocation": {
      "value": "...",
      "metadata": {
        "description": "message goes here"
      }
    }
  }
}

Reply:
Unfortunately, adding the metadata property to the parameter results in the error mentioned above when you attempt to run a filled out version of this through the cli. The same file but with the metadata property removed runs without problems. Is this expected? Will TravisCI remove these extraneous properties prior to running or runs the deployment in a different way? 

Example:

{
  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "DeploymentDataLocation" : {
      "value": "...",
      "metadata": {
        "description": "This is test value, you must specify a valid license for deployment to succeed. See the readme.md for details."
      }
    },
   ...
  }
}


Results in the error:
> az group deployment create --resource-group chlync_mercury_MBL_03 --template-file sas-viya/azuredeploy.json --verbose --parameters @/home/chlync/devel/mercury/mercury-admin/azure_test_environment/calynch/blobmirror.3.4.azuredeploy.parameters.json

Azure Error: InvalidRequestContent
Message: The request content was invalid and could not be deserialized: 'Could not find member 'metadata' on object of type 'DeploymentParameterDefinition'. Path 'properties.parameters.DeploymentDataLocation.metadata', line 1, position 48319.'.

Reply:
ok, sounds like we don't have a good option here, just remove it for now and will follow up... more ideas on this will be helpful, since users will use that as is and fail.

Reply:
For now, I have updated the file name to ThisEntireLicenseUriShouldBeReplacedWithAUriToYourOwn.zip to try and include some of what would have been in a comment.